test/README.md

```diff
@@ -0,0 +1,88 @@
+These are the tests for integration between the K frontend and the Haskell backend.
+
+`Makefile` targets:
+
+- `test`: Run the tests.
+- `golden`: Update the expected outputs.
+- `clean`
+
+Each directory contains a number of tests based on a single K definition.
+The tests require the K frontend to be installed;
+by default a suitable version will be installed at `../.build/k`.
+
+## Program tests
+
+Program tests run a concrete program with `krun`
+and compare the output to an expected ("golden") file.
+Program tests have filenames of the form `⟨name⟩.⟨ext⟩`
+where `⟨name⟩` is the (arbitrary) name of the test
+and `⟨ext⟩` is the filename extension used by the definition.
+By default, if the definition filename is `⟨def⟩.k` then `⟨ext⟩ = ⟨def⟩`,
+otherwise the `Makefile` will contain the line `EXT = ⟨ext⟩`.
+A test fails when the actual and expected output differ.
+The exit code of `krun` is not considered,
+so one may write program tests which encounter errors.
+
+`Makefile` targets:
+
+- `make ⟨name⟩.⟨ext⟩.out`:
+  Run the named test and compare the output to `⟨name⟩.⟨ext⟩.out.golden`.
+  The test passes if the expected and actual output are identical.
+  The test will not run again until the test prerequisites change
+  or `⟨name⟩.⟨ext⟩.out` is deleted.
+  If the test fails, the actual output is moved
+  from `⟨name⟩.⟨ext⟩.out` to `⟨name⟩.⟨ext⟩.out.fail`.
+- `make ⟨name⟩.⟨ext⟩.out.golden`:
+  Run the test to construct `⟨name⟩.⟨ext⟩.out.golden`.
+
+## Specification tests
```

Comment:
Not sure we should add them now, but we also have rule merge tests (https://github.com/kframework/kore/tree/master/src/main/k/working/imp/merge) and tests for saving the proof state (https://github.com/kframework/kore/tree/master/src/main/k/working/imp/save-proofs)

Reply:
Those tests are still there, but I forgot to document them. Thanks!